lib/engine/game/base.rb

```diff
@@ -564,6 +563,23 @@ def init_hexes(companies, corporations)
           end
         end
 
+        reservations = Hash.new { |k, v| k[v] = [] }
+        corporations.each do |c|
+          reservations[c.coordinates] << { entity: c,
+                                           city: c.city }
+        end
+        (corporations + companies).each do |c|
+          abilities = c.abilities(:reservation)
```

Comment:
c.abilities(:reservation) do |ability, _|
  may be cleanear

Reply:
I did try that, but `Corporation`'s override of `#abilities` is doing something I don't understand, it seems to be duplicating the abilities.

See where the `puts` block is passed, PRR's reservation gets printed twice while C&WI's just once:

```
> bundle exec rake irb
irb -I lib/ -I assets/app/
2.7.0 :001 > require 'engine'
 => true 
2.7.0 :002 > g = Engine::GAMES_BY_TITLE['1846'].new(['p1', 'p2', 'p3', 'p4', 'p5'])
2.7.0 :003 > prr = g.corporations[0]
2.7.0 :004 > cwi = g.companies[2]
2.7.0 :005 > prr.abilities(:reservation)
 => [#<Engine::Ability::Reservation:0x000055a756561000 @type=:reservation, @owner_type=nil, @when=nil, @count=nil, @used=false, @remove="4", @hex="E11", @city=0, @slot=0, @tile=<Engine::Tile: E11, hex: E11>, @owner=<Engine::Corporation: PRR>>] 
2.7.0 :006 > prr.abilities(:reservation) { |a| puts a }
#<Engine::Ability::Reservation:0x000055a756561000>
#<Engine::Ability::Reservation:0x000055a756561000>
 => [#<Engine::Ability::Reservation:0x000055a756561000 @type=:reservation, @owner_type=nil, @when=nil, @count=nil, @used=false, @remove="4", @hex="E11", @city=0, @slot=0, @tile=<Engine::Tile: E11, hex: E11>, @owner=<Engine::Corporation: PRR>>] 
2.7.0 :007 > 
2.7.0 :008 > 
2.7.0 :009 > cwi.abilities(:reservation)
 => #<Engine::Ability::Reservation:0x000055a756974908 @type=:reservation, @owner_type=nil, @when="sold", @count=nil, @used=false, @remove=nil, @hex="D6", @city=3, @slot=0, @tile=<Engine::Tile: D6, hex: D6>, @owner=<Engine::Company: C&WI>> 
2.7.0 :010 > 
2.7.0 :011 > cwi.abilities(:reservation) { |a| puts a }
#<Engine::Ability::Reservation:0x000055a756974908>
 => #<Engine::Ability::Reservation:0x000055a756974908 @type=:reservation, @owner_type=nil, @when="sold", @count=nil, @used=false, @remove=nil, @hex="D6", @city=3, @slot=0, @tile=<Engine::Tile: D6, hex: D6>, @owner=<Engine::Company: C&WI>> 

```

This output is what I would expect, so I think the problem is in `Corporation#abilities`:

```
2.7.0 :020 > prr.instance_variable_get(:@abilities)[:reservation]
 => #<Engine::Ability::Reservation:0x000055a756561000 @type=:reservation, @owner_type=nil, @when=nil, @count=nil, @used=false, @remove="4", @hex="E11", @city=0, @slot=0, @tile=<Engine::Tile: E11, hex: E11>, @owner=<Engine::Corporation: PRR>> 
```

Reply:
if the corporation owns the ability, then it will print out it's child's abilities